tests/TerminationBoxesTest.ml

```diff
@@ -82,4 +86,33 @@ let boxes = "boxes" >:::
   "./tests/widening2.c" *=> true; (* ordinals = 1 *)
   "./tests/widening3.c" => false;
   "./tests/zune.c" => false;
+
+  (* CTL termination test cases *)
+  "./tests/boolean.c" @=> true;
+  "./tests/boolean.c" @=> true;
+  "./tests/cacm2009a.c" @=> true;
+  "./tests/cav2006.c" @=> true;
+  "./tests/euclid.c" @=> false; (* conditionally terminating *)
+  "./tests/example0.c" @=> false; (* conditionally terminating *)
+  "./tests/example1.c" @=> true;
+  "./tests/example5.c" @=> false; (* conditionally terminating *)
+  "./tests/example7.c" @=> false; (* conditionally terminating *)
+  "./tests/issue8.c" @=> false; (* conditionally terminating *)
+  "./tests/postdecrement.c" @=> true;
+  "./tests/postincrement.c" @=> true;
+  "./tests/predecrement.c" @=> true;
+  "./tests/preincrement.c" @=> true;
+  "./tests/sas2010.c" @=> true;
+  "./tests/sas2014a.c" @=> false; (* conditionally terminating *)
+  "./tests/sas2014c.c" @=> false; (* conditionally terminating *)
+  "./tests/tacas2013a.c" @=> true;
+  "./tests/tap2008a.c" @=> false;
+  "./tests/tap2008b.c" @=> false;
+  "./tests/tap2008c.c" @=> false;
+  "./tests/tap2008d.c" @=> false;
+  "./tests/tap2008e.c" @=> false;
+  "./tests/tap2008f.c" @=> false;
+  "./tests/vmcai2004b.c" @=> false; (* conditionally terminating *)
+  "./tests/widening3.c" @=> false;
+  "./tests/zune.c" @=> false;
 ]
```

Comment:
We could add a function to Domain.ml to check if a tree is defined on some partitions (given by a boolean expression maybe?) and we could use that in the tests. In this way we could test more accurately the programs that only conditionally satisfy some property. What do you think?